### PR TITLE
[ExecuTorch][Core AI][5/x] Enumerated input-shape support

### PR DESCRIPTION
**Pull request by metascroy**
Adds enumerated input shapes. The partitioner resolves per input enumerations to export symbols, and preprocess substitutes them into each delegate boundary and attaches them via `set_static_shape_config`, letting a delegate carry several concrete shapes for a dynamic dim. Includes shape config tests.

### CODE REVIEW

**Comment by metascroy** [comment]
@claude review this code

File: backends/apple/coreai/compiler/enumerated_shapes.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    # Every entry has to name every boundary input, since
+    # set_static_shape_config treats a key as a whole-graph shape: an input left
+    # out is unconstrained in that specialization, and inputs sharing a symbol
+    # (a batch dim, say) would not be held equal.
+    boundary_inputs = []
+    active_symbols = set()
+    for uname in user_inputs:
+        # Match edge user inputs to coreai graph inputs by name; skip anything
+        # that isn't a coreai tensor input (e.g. a taken-over constant).
+        if not isinstance(uname, str) or uname not in coreai_names:
```
Comment (metascroy) [inline backends/apple/coreai/compiler/enumerated_shapes.py:204]: @cymbalrush does CoreAI preserve user name from torch.export in its graph_input_names?

It appears to from the passing tests, but I don't know if this is a contract you have

**Comment by metascroy** [comment]
@claude review this code

**Review by SS-JIA — APPROVED** [review]
